hive-metastore/src/main/java/org/apache/iceberg/hive/HMSTablePropertyHelper.java

```diff
@@ -0,0 +1,264 @@
+    } else {
+      parameters.remove(hive_metastoreConstants.META_TABLE_STORAGE);
+    }
+  }
+
+  @VisibleForTesting
+  static void setSnapshotStats(
+      TableMetadata metadata, Map<String, String> parameters, long maxHiveTablePropertySize) {
+    parameters.remove(TableProperties.CURRENT_SNAPSHOT_ID);
+    parameters.remove(TableProperties.CURRENT_SNAPSHOT_TIMESTAMP);
+    parameters.remove(TableProperties.CURRENT_SNAPSHOT_SUMMARY);
+
+    Snapshot currentSnapshot = metadata.currentSnapshot();
+    if (exposeInHmsProperties(maxHiveTablePropertySize) && currentSnapshot != null) {
+      parameters.put(
+          TableProperties.CURRENT_SNAPSHOT_ID, String.valueOf(currentSnapshot.snapshotId()));
+      parameters.put(
+          TableProperties.CURRENT_SNAPSHOT_TIMESTAMP,
+          String.valueOf(currentSnapshot.timestampMillis()));
+      setSnapshotSummary(parameters, currentSnapshot, maxHiveTablePropertySize);
+    }
+
+    parameters.put(TableProperties.SNAPSHOT_COUNT, String.valueOf(metadata.snapshots().size()));
+  }
+
+  @VisibleForTesting
+  static void setSnapshotSummary(
+      Map<String, String> parameters, Snapshot currentSnapshot, long maxHiveTablePropertySize) {
+    try {
+      String summary = JsonUtil.mapper().writeValueAsString(currentSnapshot.summary());
+      if (summary.length() <= maxHiveTablePropertySize) {
+        parameters.put(TableProperties.CURRENT_SNAPSHOT_SUMMARY, summary);
+      } else {
+        LOG.warn(
+            "Not exposing the current snapshot({}) summary in HMS since it exceeds {} characters",
+            currentSnapshot.snapshotId(),
+            maxHiveTablePropertySize);
+      }
+    } catch (JsonProcessingException e) {
+      LOG.warn(
+          "Failed to convert current snapshot({}) summary to a json string",
+          currentSnapshot.snapshotId(),
+          e);
+    }
+  }
+
+  @VisibleForTesting
+  static void setPartitionSpec(
+      TableMetadata metadata, Map<String, String> parameters, long maxHiveTablePropertySize) {
+    parameters.remove(TableProperties.DEFAULT_PARTITION_SPEC);
+    if (exposeInHmsProperties(maxHiveTablePropertySize)
+        && metadata.spec() != null
+        && metadata.spec().isPartitioned()) {
+      String spec = PartitionSpecParser.toJson(metadata.spec());
+      setField(parameters, TableProperties.DEFAULT_PARTITION_SPEC, spec, maxHiveTablePropertySize);
+    }
+  }
+
+  @VisibleForTesting
+  static void setSortOrder(
+      TableMetadata metadata, Map<String, String> parameters, long maxHiveTablePropertySize) {
+    parameters.remove(TableProperties.DEFAULT_SORT_ORDER);
+    if (exposeInHmsProperties(maxHiveTablePropertySize)
+        && metadata.sortOrder() != null
+        && metadata.sortOrder().isSorted()) {
+      String sortOrder = SortOrderParser.toJson(metadata.sortOrder());
+      setField(parameters, TableProperties.DEFAULT_SORT_ORDER, sortOrder, maxHiveTablePropertySize);
+    }
+  }
+
+  public static void setSchema(
```

Comment:
Could this be package-private with `@VisibleForTesting` annotation?

Reply:
Sure.